floss/qs/main.py

```diff
@@ -591,6 +617,29 @@ class Layout(abc.ABC):
     # so they come from before/between/after the children ranges.
     strings: List[TaggedString] = field(init=False, default_factory=list)
 
+    def to_dict(self) -> Dict:
+        d = {
+            "name": self.name,
+            "offset": self.offset,
+            "size": self.slice.range.length,
+            "children": [child.to_dict() for child in self.children],
+        }
+        return d
```

Comment:
we might want to rely on Pydantic to do this (de)serialization for us.

I'm not sure I like the JSON not matching the Python dataclasses. I'd suggest the nesting and structure be the same so its easier to map JS to/from Python.

Reply:
so should we move from dataclass to pydantic dataclass/BaseModel?

Reply:
on it, I'm also reconsidering using your originally proposed structure where each layout stores its strings nested under itself

pydanti by default results in a quite verbose serialization (with all the the current classes, e.g., slice + range) so maybe we want to fine tune that - or not :)